### ISSUE TITLE
[BUG]Phi-3.5-Mini Genie API service exits immediately; LLaMA-3.2-1B fails with DMA Error 1100 on Snapdragon X Elite

### ISSUE THREAD

**Issue by Ambarayya** [qualcomm/ai-hub-apps#128]
**Environment**
Platform: Windows on Snapdragon (Snapdragon X Elite)
Genie API Service: C++ GenieAPIService 2.0.0
Genie Library: 1.12.0
Backend: QNN HTP
Execution Method: C++ Genie API Service (BAT file) and genie-t2t-run
Python GenieAPIService: Not usable (QNN system interface error)

**Models Tested**
IBM-Granite-v3.1-8B-Instruct (works)
LLaMA-3.2-1B (fails with DMA error)
Phi-3.5-Mini  (service exits)

**Issue 1: Phi-3.5-Mini Genie API service exits immediately (no visible error)**
Description
When starting the C++ GenieAPIService with the Phi-3.5-Mini model, the service:
Loads the model successfully
Allocates buffers
Starts the HTTP server
Prints “Genie API Service IS Running”
Then exits immediately and returns to the command prompt
This happens without any explicit error message.

**Steps to Reproduce**
4.Start_GenieAPIService.bat Phi-3.5-mini

**Observed Logs (excerpt)**
[INFO] "Allocated total size = 816947712 across 8 buffers"
[W] Model loaded: Phi-3.5-mini
[A] [OK] Genie API Service IS Running.
[A] [OK] Genie API Service -> http://127.0.0.1:8910
Press any key to continue . . .

**Expected Behavior**

The Genie API service should remain running and continue listening for API requests, similar to IBM-Granite  models.

**Issue 2: LLaMA-3.2-1B fails with DMA Error 1100 when using Genie API**
**Description**
When running LLaMA-3.2-1B using the Genie API service, the model fails during execution with a DMA execution error (Error 1100).
**Steps to Reproduce**
4.Start_GenieAPIService.bat Llama3.2-1B

**Observed Error**
<E> Dma execution failed on the skel side. result = 1100
<E> Graph ar128_cl4096_2_of_3 failed in execution with err 1100
[ERROR] Failed to execute graph. Error 1100
Engine returned 0 logits.

**Questions**

1. Is Phi-3.5-Mini expected to require special prompt.json / instruct configuration to keep the C++ Genie API service alive?
2. Is the immediate exit of the service a known behavior for Phi-3.5-Mini?
3. Is DMA Error 1100 for LLaMA-3.2-1B possibly related to::
Context length (4096)
Multi-graph execution
Or device-specific limitations on Snapdragon X Elite?
